### PR TITLE
Prepare active_users_aggregates for a backfill with shredder mitigation.

### PR DESCRIPTION
**Pull request by lucia-vargas-a**
## Description
This PR prepares active_users_aggregates for a backfill with shredder mitigation, which includes renaming columns to cascade upstream changes.

The full backfill from 2018-01-01 until 2024-11-10 is available in `moz-fx-data-shared-prod.backfills_staging_derived.firefox_desktop_derived__active_users_aggregates_v4_2024_11_15`.

## Related Tickets & Documents
* [DENG-4060](https://mozilla-hub.atlassian.net/browse/DENG-4060)

**Reviewer, please follow [this checklist](https://github.com/mozilla/bigquery-etl/blob/main/.github/reviewer_checklist.md)**

[DENG-4060]: https://mozilla-hub.atlassian.net/browse/DENG-4060?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-5490)

### CODE REVIEW

File: sql_generators/active_users_aggregates_v4/__init__.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+            )
+            schema_template = desktop_schema_template
+        elif browser.name == "focus_android":
+            query_sql = reformat(
+                focus_android_query_template.render(
+                    project_id=target_project,
+                    app_name=browser.name,
+                )
+            )
+            schema_template = mobile_schema_template
```
Comment (bochocki) [inline sql_generators/active_users_aggregates_v4/__init__.py:125]: Why does focus_android have a query template that's distinct from the mobile query template?

File: sql_generators/active_users_aggregates_v3/templates/mobile_schema.yaml [hunk 2]
```diff
@@ -2,84 +2,92 @@ fields:
...
+  description: Classification of client_ids based on usage and active state.
+- name: app_name
   type: STRING
   mode: NULLABLE
-- name: attribution_source
+  description: Browser name.
+- name: app_version
   type: STRING
   mode: NULLABLE
-- name: attributed
```
Comment (bochocki) [inline sql_generators/active_users_aggregates_v3/templates/mobile_schema.yaml:14]: attribution_medium, attribution_source, and attributed all seem to be removed from the schema. Is that intentional?

File: sql_generators/active_users_aggregates_v4/templates/mobile_schema.yaml [hunk 3]
```diff
@@ -0,0 +1,93 @@
+fields:
```
Comment (bochocki) [inline sql_generators/active_users_aggregates_v4/templates/mobile_schema.yaml:1]: The desktop v4 schema contains attribution columns while the mobile schema doesn't; is that intentional?

**Review by bochocki — COMMENTED** [review]
I have some comments that I think are worth addressing, but it's possible that they're not very relevant. If we can address the comments this looks good to go.

File: sql_generators/active_users_aggregates_v4/__init__.py [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+from bigquery_etl.cli.utils import use_cloud_function_option
+from bigquery_etl.format_sql.formatter import reformat
+from bigquery_etl.util.common import render, write_sql
+
+THIS_PATH = Path(os.path.dirname(__file__))
+TABLE_NAME = os.path.basename(os.path.normpath(THIS_PATH))
+BASE_NAME = "_".join(TABLE_NAME.split("_")[:-1])
+DATASET_FOR_UNIONED_VIEWS = "telemetry"
+CHECKS_TEMPLATE_CHANNELS = {
```
[the remark this answers is not in the document]
Comment (lucia-vargas-a) [reply sql_generators/active_users_aggregates_v4/__init__.py:18]: Yes it is expected. Fenix checks are in a separate file.

File: sql_generators/active_users_aggregates_v3/templates/mobile_schema.yaml (hunk 2 above)
Comment (lucia-vargas-a) [reply sql_generators/active_users_aggregates_v3/templates/mobile_schema.yaml:14]: Not intentional, added back, thanks!

File: sql_generators/active_users_aggregates_v3/templates/metadata.yaml [hunk 5]
```diff
@@ -22,9 +26,10 @@ owners:
 labels:
   incremental: true
   change_controlled: true
+  shredder_mitigation: true
 scheduling:
   dag_name: bqetl_analytics_aggregations
-  task_name: {{ app_name }}_active_users_aggregates
+  task_name: {{ app_name }}_active_users_aggregates_v3
```
[the remark this answers is not in the document]
Comment (lucia-vargas-a) [reply sql_generators/active_users_aggregates_v3/templates/metadata.yaml:32]: You;re right that the task must be named _v4, this happens  in the folder for version 4.
In v3 folder the task remains as v3 and pending to be removed from the DAG once the v4 is in production.

File: sql_generators/active_users_aggregates_v4/__init__.py (hunk 1 above)
Comment (lucia-vargas-a) [reply sql_generators/active_users_aggregates_v4/__init__.py:125]: It was the only browser that required keeping legacy and Glean data, probably something to reevaluate soon.

File: sql_generators/active_users_aggregates_v3/templates/desktop_query.sql [hunk 6]
```diff
@@ -7,7 +7,7 @@ WITH todays_metrics AS (
     app_version AS app_version,
     normalized_channel AS channel,
     IFNULL(country, '??') country,
```
Comment (bochocki) [inline sql_generators/active_users_aggregates_v3/templates/desktop_query.sql:9]: nitpicky, but for consistency:
```suggestion
    IFNULL(country, '??') AS country,
```

File: sql_generators/active_users_aggregates_v3/templates/desktop_query.sql [hunk 7]
```diff
@@ -7,7 +7,7 @@ WITH todays_metrics AS (
...
-    city,
+    IFNULL(city, '??') city,
```
Comment (bochocki) [inline sql_generators/active_users_aggregates_v3/templates/desktop_query.sql:10]:
```suggestion
    IFNULL(city, '??') AS city,
```

**Review by bochocki — APPROVED** [review]
Some very minor nits. Looks great though, thanks Lucia!

File: sql_generators/active_users_aggregates_v3/templates/desktop_query.sql (hunk 6 above)
Comment (lucia-vargas-a) [reply sql_generators/active_users_aggregates_v3/templates/desktop_query.sql:9]: Thanks!

**Pull request merged**